httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/ContentCompressionExec.java

```diff
@@ -70,16 +71,20 @@
 @Internal
 public final class ContentCompressionExec implements ExecChainHandler {
 
-    private final String[] acceptEncoding;
+    private final Header acceptEncoding;
     private final Lookup<InputStreamFactory> decoderRegistry;
     private final boolean ignoreUnknown;
 
     public ContentCompressionExec(
             final List<String> acceptEncoding,
             final Lookup<InputStreamFactory> decoderRegistry,
             final boolean ignoreUnknown) {
-        this.acceptEncoding = acceptEncoding != null ? acceptEncoding.toArray(
-                new String[acceptEncoding.size()]) : new String[] {"gzip", "x-gzip", "deflate"};
+        // MessageSupport.format is not thread-safe, because it sorts the
```

Comment:
@coxlinton I will fix `MessageSupport#format` in HttpCore master. Do you mind dropping the comment about `MessageSupport#format` thread safety from this change-set?

Reply:
Not at all - that definitely seems like an excellent idea. Updated made.

Reply:
@coxlinton Could you please squash those two commits into a single one?

Reply:
Sorry about that - done.